components/authz-service/storage/v2/conditionattribute.go

```diff
@@ -0,0 +1,76 @@
+package v2
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ConditionAttribute is an enum of attributes a project rule condition can be.
+type ConditionAttribute int
+
+const (
+	Organization ConditionAttribute = iota
+	ChefServer
+	Environment
+	ChefRole
+	ChefTag
+	PolicyName
+	PolicyGroup
+)
+
+func (c ConditionAttribute) String() string {
+	if c < Organization || c > PolicyGroup {
+		panic(fmt.Sprintf("unknown value from iota ConditionAttribute on String() conversion: %d", c))
+	}
+
+	return conditionAttributeStringValues()[c]
+}
+
+// NewConditionAttribute converts a string to an ConditionAttribute or returns an error.
+func NewConditionAttribute(in string) (ConditionAttribute, error) {
+	switch in {
+	case "organization":
```

Comment:
I wonder if it's worth pulling these into a hash (instead of `conditionAttributeStringValues`) and then checking for the keys/returning the node, erring if that doesn't succeed. It would get us closer to having some constants around the strings. 💭

Reply:
We want conditionAttributeStringValues to be a slice because that's how we convert between the enum and string (the enum is composed of integers that corresponds to slice indices).